### PR TITLE
Add detail to doc for add_surface_from_arrays

### PR DESCRIPTION
**Pull request by garyo**
I'm no expert at this XML doc format, so please let me know if there are any formatting or content issues and I'll fix them.

### CODE REVIEW

**Review by cbscribe — CHANGES_REQUESTED** [review]
Thanks for your contribution. I've made some suggestions for clarity and to follow the docs guidelines.

**Comment by garyo** [comment]
Updated per your comments, @cbscribe

File: doc/classes/ArrayMesh.xml [hunk 1]
```diff
@@ -29,8 +29,12 @@
...
+				The [code]arrays[/code] argument is an array of arrays. See [enum ArrayType] for the values used in this array. For example, [code]arrays[0][/code] is the array of vertices. That first vertex sub-array is always required; the others are optional. Adding an index array puts this function into "index mode" where the vertex and other arrays become the sources of data and the index array defines the vertex order. All sub-arrays must have the same length as the vertex array or be empty, except for [code]ARRAY_INDEX[/code] if it is used.
+				Adding an index array puts this function into "index mode" where the vertex and other arrays become the sources of data, and the index array defines the order of the vertices.
+                Godot uses clockwise winding order for front faces of triangle primitive modes.
```
Comment (Noshyaar) [inline doc/classes/ArrayMesh.xml]: Indentation issue.

File: doc/classes/ArrayMesh.xml [hunk 2]
```diff
@@ -29,8 +29,12 @@
 			<argument index="3" name="compress_flags" type="int" default="97792">
 			</argument>
 			<description>
-				Create a new surface ([method get_surface_count] that will become surf_idx for this.
-				Surfaces are created to be rendered using a "primitive", which may be PRIMITIVE_POINTS, PRIMITIVE_LINES, PRIMITIVE_LINE_STRIP, PRIMITIVE_LINE_LOOP, PRIMITIVE_TRIANGLES, PRIMITIVE_TRIANGLE_STRIP, PRIMITIVE_TRIANGLE_FAN. (As a note, when using indices, it is recommended to only use just points, lines or triangles).
+				Creates a new surface.
+
```
Comment (Noshyaar) [inline doc/classes/ArrayMesh.xml]: There should be no empty line. One line is one paragraph.

**Comment by garyo** [comment]
Updated per your comments, @Noshyaar.

**Comment by hpvb** [comment]
Cherry picked into 3.0.3

**Pull request merged**